src/vs/editor/browser/viewParts/lineNumbers/lineNumbers.ts

```diff
@@ -190,3 +190,13 @@ export class LineNumbersOverlay extends DynamicViewOverlay {
 		return this._renderResult[lineIndex];
 	}
 }
+
+registerThemingParticipant((theme, collector) => {
+	const editorLineNumbersColor = theme.getColor(editorLineNumbers);
+	const editorDimmedLineNumberColor = theme.getColor(editorDimmedLineNumber);
+	if (editorDimmedLineNumberColor) {
+		collector.addRule(`.monaco-editor .line-numbers.dimmed-line-number { color: ${editorDimmedLineNumberColor}; }`);
+	} else if (editorLineNumbersColor) {
+		collector.addRule(`.monaco-editor .line-numbers.dimmed-line-number { color: ${editorLineNumbersColor.transparent(0.4)}; }`);
+	}
```

Comment:
The first could move to css easily, I don't know of a good way for the second unless it doesn't back a background color set in which case you can use `opacity: 0.4`

Reply:
I thought about using opacity, but didn't go with that because that can't be configured via a color customization / theme.

I agree about the first rule could be written in CSS. I kept it here because of the `else if`. I could have written the code better e.g.:

```
const editorDimmedLineNumberColor = theme.getColor(editorDimmedLineNumber) || editorLineNumbersColor.transparent(0.4);
```